### ISSUE TITLE
data isn't in the correct format for URLs with non-ASCII code

### ISSUE THREAD

**Issue by freewizard** [metabolist/metatext#25]
**Describe the bug**
When URL field in opengraph card or search hash tags contains non-ASCII code, which is common in pleroma or other non-Mastodon instances, e.g. this response of search api:
`{"accounts":[],"hashtags":[{"name":"","url":"https://pleroma.site/tag/标签"}],"statuses":[]}`
or this in response of timeline api:
```
	    "card": {
	        "description": "---",
	        "image": null,
	        "pleroma": {
	            "opengraph": {
	                "card": "summary",
	                "description": "---",
	                "image": "",
	                "site_name": "Telegraph",
	                "title": "---",
	                "type": "article",
	                "url": "https://telegra.ph/タグ"
	            }
	        },
	        "provider_name": "telegra.ph",
	        "provider_url": "https://telegra.ph",
	        "title": "---",
	        "type": "link",
	        "url": "https://telegra.ph/タグ"
	    },
```
this error will occur: 
`The data couldn't be read because it isn't in the correct format.`

**To Reproduce**
1. Go to Explore tab of any pleroma site
2. Click on Hashtags
3. Type 标签
4. See error `The data couldn't be read because it isn't in the correct format.`

**Expected behavior**
search result should be properly parsed.

**Screenshots**

**(please complete the following information):**
 - iPhone 12
 - iOS 14.4.1
 - App Version 1.0.2(5)

**Additional context**

**Comment by jzzocc** [comment]
Ah yes, this is because Swift's `URL` struct needs non-ASCII characters to be percent encoded in the string it's constructed from. I currently do this conversion only for custom emoji URLs since it was the only place I encountered the issue, but it looks like It'd probably be better to do it for all URLs in the API. Thanks for pointing this out.

**Comment by jzzocc** [comment]
This will be fixed in v1.1.1
